### DISCUSSION TITLE
Json schema for annotated field with plainvalidator

### DISCUSSION THREAD

**Discussion by orfisko** [pydantic/pydantic#9762] [Question] [answered]
I am struggling to find in the docs how I need to allow the generation of a JSON schema for a field defined like this:

```python 
def validate_phone_number(value: str) -> str:
    pattern = re.compile(r"^[+][(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$")
    if not pattern.match(value):
        raise ValueError('Invalid phone number format. should look like +32494879426')
    return value

TelephoneNumber = Annotated[str, PlainValidator(validate_phone_number)]
```

If I use TelephoneNumber as a field in a basemodel the validator works and gets called etc, so all good there. But when a JSON schema needs to be created of a model which uses this field, things go haywire with error messages like this:

pydantic.errors.PydanticInvalidForJsonSchema: Cannot generate a JsonSchema for core_schema.PlainValidatorFunctionSchema ({'type': 'no-info', 'function': <function validate_phone_number at 0x7fee04c709a0>})

How can I fix this or add this info? I don't want to add specific documentation to the phone number in the json schema, but the schema as a whole should work.

**Comment by orfisko** [comment, marked as answer]
Found it [here](https://docs.pydantic.dev/latest/concepts/json_schema/#withjsonschema-annotation)

**Comment by Viicos** [comment]
Note that https://github.com/pydantic/pydantic/issues/8208 might improve this in the near future.
